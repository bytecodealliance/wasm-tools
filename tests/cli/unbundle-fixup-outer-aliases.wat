;; RUN: component unbundle --module-dir %tmpdir --wat % --threshold 0

(component $C
  (core type $t (func))
  (core type $m (module
    (alias outer $C $t (type))
  ))

  (component $C2
    (alias outer $C $t (core type))
    (core type $t (func))
    (core type $m (module
      (alias outer $C2 $t (type))
    ))
  )

  (core module $m
    (func $f)
    (export "x" (func $f))
  )

  (alias outer $C $m (core module $m2))

  (component
    (alias outer $C $m (core module))
    (alias outer $C $m2 (core module))
  )
)


