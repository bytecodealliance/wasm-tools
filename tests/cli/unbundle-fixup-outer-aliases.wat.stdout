(component $C
  (core type (;0;) (func))
  (core type (;1;)
    (module
      (alias outer $C 0 (type (;0;)))
      (export "x" (func (type 0)))
    )
  )
  (import "unbundled-module0" (core module (;0;) (type 1)))
  (core type $t (;2;) (func))
  (core type $m (;3;)
    (module
      (alias outer $C $t (type (;0;)))
    )
  )
  (component $C2 (;0;)
    (alias outer $C $t (core type (;0;)))
    (core type $t (;1;) (func))
    (core type $m (;2;)
      (module
        (alias outer $C2 $t (type (;0;)))
      )
    )
  )
  (alias outer $C 0 (core module $m (;1;)))
  (alias outer $C $m (core module $m2 (;2;)))
  (component (;1;)
    (alias outer $C $m (core module (;0;)))
    (alias outer $C $m2 (core module (;1;)))
  )
)
